File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    int cents;
    int reszta;
    int quaters;
    int dimes;
    int nickels;
    int pennies;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);
    cents = round(dollars * 100);
    quaters = floor(cents / 25);
    dimes = floor((cents - quaters * 25) / 10);
    nickels = floor((cents - quaters * 25 - dimes * 10) / 5);
    pennies = floor((cents - quaters * 25 - dimes * 10 - nickels * 5) / 1);
    reszta = quaters + dimes + nickels + pennies;
    printf("%d\n", reszta);
}

File: pset6/cash/cash.py
from cs50 import get_float

while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:
        break

cents = dollars * 100
quaters = cents // 25
cents -= quaters * 25
dimes = cents // 10
cents -= dimes * 10
nickels = cents // 5
cents -= nickels * 5
pennies = cents // 1

print(int(quaters + dimes + nickels + pennies))
